components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -3396,6 +3414,94 @@ bool RewardsServiceImpl::IsRewardsEnabled() const {
   return false;
 }
 
+bool RewardsServiceImpl::GetScheduledCaptchaInfo(std::string* url,
+                                                 bool* max_attempts_exceeded) {
+  DCHECK(url);
+  DCHECK(max_attempts_exceeded);
+
+  const std::string payment_id =
+      profile_->GetPrefs()->GetString(prefs::kScheduledCaptchaPaymentId);
+  const std::string captcha_id =
+      profile_->GetPrefs()->GetString(prefs::kScheduledCaptchaId);
+  if (payment_id.empty() || captcha_id.empty()) {
+    return false;
+  }
+
+  const int failed_attempts =
+      profile_->GetPrefs()->GetInteger(prefs::kScheduledCaptchaFailedAttempts);
+
+#if BUILDFLAG(BRAVE_ADAPTIVE_CAPTCHA_ENABLED)
```

Comment:
why would this method be expose at all if BRAVE_ADAPTIVE_CAPTCHA_ENABLED is not enabled? It does nothing in that case

Reply:
Good point, fixed.